File: src/main/java/ac/kr/deu/connect/luck/event/repository/EventRepository.java
package ac.kr.deu.connect.luck.event.repository;

import ac.kr.deu.connect.luck.event.entity.EventStatus;
import ac.kr.deu.connect.luck.event.entity.Event;
import ac.kr.deu.connect.luck.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface EventRepository extends JpaRepository<Event, Long> {

    void deleteByManager(User manager);

    List<Event> findAllByStatus(EventStatus status);

    List<Event> findAllByManager(User manager);
}

File: src/main/java/ac/kr/deu/connect/luck/event/repository/EventRepository.kt
package ac.kr.deu.connect.luck.event.repository

import ac.kr.deu.connect.luck.event.entity.Event
import ac.kr.deu.connect.luck.event.entity.EventProvider
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface EventRepository : JpaRepository<Event, Long> {
    fun findByManagerId(
        managerId: Long,
        pageable: Pageable,
    ): Page<Event>


    fun findByProvider(provider: EventProvider): Event?
}
